data_safe_haven/administration/users/guacamole_users.py

```diff
@@ -3,18 +3,21 @@
 from typing import Any
 
 from data_safe_haven.config import Config
+from data_safe_haven.config.context_settings import Context
```

Comment:
As a rule-of-thumb (can't find a reference to it at the moment) it's better not to use deeply-nested imports in Python code. This makes your dependencies more fragile and means that it's harder to refactor. In this case, I think we'd be better with the following:

```suggestion
from data_safe_haven.config import Config, Context
```

with an equivalent change to `config > __init__.py`.

Reply:
I did this because there were circular dependencies.

I think that might not be a problem any more so I'll look to tidy that.

Reply:
For local dependencies I prefer `from .context_settings import Context`. If they're really circular, we should think about how to break the coupling.

Reply:
Done in 2012ff6